access_tag.go

```diff
@@ -0,0 +1,108 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/goccy/go-json"
+)
+
+type AccessTag struct {
+	Name     string `json:"name,omitempty"`
+	AppCount int    `json:"app_count,omitempty"`
+}
+
+type AccessTagListResponse struct {
+	Response
+	Result     []AccessTag `json:"result"`
+	ResultInfo `json:"result_info"`
+}
+
+type AccessTagResponse struct {
+	Response
+	Result AccessTag `json:"result"`
+}
+
+type ListAccessTagsParams struct{}
+
+type CreateAccessTagParams struct {
+	Name string `json:"name,omitempty"`
+}
+
+type UpdateAccessTagParams struct {
+	Name string `json:"name,omitempty"`
+}
+
+func (api *API) ListAccessTags(ctx context.Context, rc *ResourceContainer, params ListAccessTagsParams) ([]AccessTag, error) {
+	uri := buildURI(fmt.Sprintf("/%s/%s/access/tags", rc.Level, rc.Identifier), params)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return []AccessTag{}, err
+	}
+
+	var TagsResponse AccessTagListResponse
+	err = json.Unmarshal(res, &TagsResponse)
+	if err != nil {
+		return []AccessTag{}, err
+	}
+	return TagsResponse.Result, nil
+}
+
+func (api *API) GetAccessTag(ctx context.Context, rc *ResourceContainer, tagName string) (AccessTag, error) {
+	uri := fmt.Sprintf("/%s/%s/access/tags/%s", rc.Level, rc.Identifier, tagName)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return AccessTag{}, err
+	}
+
+	var TagResponse AccessTagResponse
+	err = json.Unmarshal(res, &TagResponse)
+	if err != nil {
+		return AccessTag{}, err
+	}
+	return TagResponse.Result, nil
+}
+
+func (api *API) CreateAccessTag(ctx context.Context, rc *ResourceContainer, params CreateAccessTagParams) (AccessTag, error) {
+	uri := fmt.Sprintf("/%s/%s/access/tags", rc.Level, rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
+	if err != nil {
+		return AccessTag{}, err
+	}
+
+	var TagResponse AccessTagResponse
+	err = json.Unmarshal(res, &TagResponse)
+	if err != nil {
+		return AccessTag{}, err
+	}
+	return TagResponse.Result, nil
+}
+
+func (api *API) DeleteAccessTag(ctx context.Context, rc *ResourceContainer, tagName string) error {
+	uri := fmt.Sprintf("/%s/%s/access/tags/%s", rc.Level, rc.Identifier, tagName)
+	_, err := api.makeRequestContext(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (api *API) UpdateAccessTag(ctx context.Context, rc *ResourceContainer, params UpdateAccessTagParams) (AccessTag, error) {
```

Comment:
this does something however, i don't quite see what it is changing. the `params.Name` passed in is used for the old and new value. should there be a difference available to the values?

Reply:
we're gonna have to do something hacky to make this work because I'm realizing with TF you're not gonna be specifying the url you're making the request to so we need to differentiate between the name that is the current name of the tag and what the updated name will be since the name of the tag is also its unique identifier in the api url

Reply:
In retrospect we're gonna get rid of the update functionality on tags because you would also have to update all the tags in the apps that have said tags anyway (the api does this inherently but that doesn't fix tf state)

Reply:
it won't just be terraform that is impacted here, it will also be the SDKs. reason being is that we are using a modifiable value in the URL and the value to update. i don't recommend this particular API design but since we have it, the `Update` method can have `TagName` and `UpdatedTagName` to differentiate the two.

Reply:
updated to make the update function a noop